docs/modules/ROOT/pages/explanations/decisions/capacity-alerting.adoc

```diff
@@ -0,0 +1,111 @@
+This can be solved by https://github.com/vshn/espejo[espejo] or through a custom controller, but we need to be aware that increase the potential attack surface by distributing Opsgenie credentials to many namespaces.
+
+=== Expose User-Workload Monitoring
+
+APPUiO Cloud and APPUiO Managed OpenShift come with a versatile user-workload monitoring system that allows a user to specify their own alert-manager configuration on a namespace level.
+
+We could expose the alert manager configuration of this system directly in the database claim and the composition will then create an `AlertManagertConfig` resource in the database namespace.
+This would enable a service user to decide for themselves how they want to be alerted about capacity issues.
+
+With such a system the database claim could look similar to the following, which would configure all alerts to be sent to the service users slack instance.
+
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: pgsql-app1-prod
+  namespace: prod-app
+spec:
+  parameters:
+    alerting:
+      receivers:
+      - name: slack
+        slackConfigs:
+        - apiURL:
+            key: url
+            name: slack-alert-webhook-secret
+          channel: '#alerts'
+      route:
+        groupBy: [alertname]
+        receiver: slack
+  writeConnectionSecretToRef:
+    name: postgres-creds
+----
+
+=== Send E-Mail Alert through Alert-Manager and mailgun
+
+VSHN has the option to send e-mails automatically through https://www.mailgun.com/[mailgun].
+This can be leveraged to inform service users through e-mail without a lot of setup overhead.
+
+In practice this can be implemented fairly similarly to forwarding alerts to Opsgenie.
+The composition will create an `AlertManagertConfig` resource in the database namespace that configures alert-manager to send mails to the provided e-mail address.
+
+With such a system the database claim could look similar to the following, which would send all alerts as mails to `oncall@example.com`
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: pgsql-app1-prod
+  namespace: prod-app
+spec:
+  parameters:
+    alerting:
+      email: oncall@example.com
+  writeConnectionSecretToRef:
+    name: postgres-creds
+----
+
+There are a few things we'll need to consider for such an implementation:
+
+* This setup forces us to replicate a mailgun token to every database namespace which increases the potential attack surface.
+* We might need to think about verifying the e-mail before forwarding alerts to it.
+This could be done through a custom controller.
+
+== Decision
+
+We implement both exposing the user-workload monitoring and automated e-mail alerts through mailgun.
+E-mail based alerting should be mandatory to ensure that we can prove that the service user was informed of the issue.
```

Comment:
Not sure yet about that decision. At least with "should" we have some wiggle room for figuring out the exact implementation.

Reply:
Perfect, thanks.